### DISCUSSION TITLE
The consequences of 'draft' on existing binaries

### DISCUSSION THREAD

**Discussion by sagantech** [free-audio/clap#306] [Q&A] [answered]
If hypothetically Acme corporation releases CLAP host v1.0 with a complete implementation of a CLAP extension that is marked as draft ; such as preset load:

 CLAP_EXT_PRESET_LOAD[] = "clap.preset-load.draft/1";

Then subsequently the draft extension is implemented into the CLAP spec, and is no longer draft, is the Acme CLAP host going to lose its functionality?

**Comment by abique** [comment, marked as answer]
This is a good question, if the interface doesn't change we could actually keep the extension id the same or let the host/plugin knows that the draft has the same ABI.

I agree that we have some extensions which have been marked as draft for a long time, but they need to be tested in production and the release cycles for bitwig or even u-he plugins are a bit slow so we're doing our best but it takes time.

**Comment by robbert-vdh** [comment]
> Then subsequently the draft extension is implemented into the CLAP spec, and is no longer draft, is the Acme CLAP host going to lose its functionality?

They could choose to keep support for the draft extension around, but realistically the draft versions should be considered drafts and should no longer be used by plugins and hosts once the extension has been finalized or a new draft has been released. In theory this should not cause too many issues since extensions that are implemented by more than a couple plugins and/or hosts and are deemed ready for prime time can be moved out of draft.
